File: algorithm/src/main/java/baekjoon/p05586/Main.java
package baekjoon.p05586;

import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        for (int solve : solve(sc.next())) {
            System.out.println(solve);
        }
    }

    private static int[] solve(String input) {
        int[] ret = new int[2];

        for (int i = 0; i <= input.length() - 3; i++) {
            switch (input.substring(i, i + 3)) {
                case "JOI":
                    ret[0]++;
                    break;
                case "IOI":
                    ret[1]++;
                    break;
            }
        }

        return ret;
    }

}

File: algorithm/src/main/java/baekjoon/p05586/Main.kt
package baekjoon.p05586

fun main(args: Array<String>) {
    println(solve(readLine()!!).joinToString("\n"))
}

private fun solve(input: String): Array<Int> {
    val list = (0..input.lastIndex - 2).map { input.substring(it, it + 3) }
    return arrayOf(
            list.count { it == "JOI" },
            list.count { it == "IOI" }
    )
}